app/controllers/words_controller.rb

```diff
@@ -0,0 +1,20 @@
+class WordsController < ApplicationController
+  def index
+    @title = t "title.words_index"
+    load_words_index
+    @courses = Course.all.order("name")
+    @courses_name = @courses.map(&:name)
+  end
+
+  def load_words_index
+    @words = Word.joins(:course).all.order("words.name")
+    @words = @words.name_like(params[:alphabe]) if params[:alphabe].present?
+    @words = @words.course_name(params[:course]) if params[:course].present?
```

Comment:
Code này lặp nhỉ? tìm cách viết lại nào, cơ mà:
+ có khi nào tồn tại cả 2 điều kiện search là alphabe và course ko chú?
+ việc kiểm tra 2 param này có tồn tại ko, chú có thể viết ở model nhé, để khỏi check present? ở đây
kiểu như : `scope :course_name, ->(params){where("courses.name = ?", params) if params[:course].present?}

Reply:
về logic thì có đk search nào thì search cái đó, có cả 2 thì search liên tiếp a and b.
về cách viết để em sửa lại, em cảm ơn chị nhiều nà...